tests/search_tree.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import Generic, Optional, Protocol, TypeVar
+
+
+class SelfComparable(Protocol):
+    def __lt__(self: T, other: T) -> bool:
+        ...
+
+
+T = TypeVar("T", bound=SelfComparable)
+
+
+class SearchTree(Generic[T]):
+    def __init__(self) -> None:
+        self.root: Optional[Node[T]] = None
+
+    def insert(self, value: T) -> None:
+        if self.root is None:
+            self.root = Node(value=value)
+            return
+        node = self.root
+        while True:
+            if node.value < value:
```

Comment:
Aren't the smaller nodes usually placed on the *left* of the current node? "If the value is smaller than the current node, move left" (https://courses.engr.illinois.edu/cs225/sp2022/resources/bst/)